File: hw/bonilla/U3/HW17Singleton/Singletonjs/USTax.js
class USTax {
  static #instance;
  static #TAXRATE = 0.15;

  constructor() {
    if (USTax.#instance) {
      throw new Error("Use USTax.getInstance() to get the singleton instance.");
    }
  }

  static getInstance() {
    if (!USTax.#instance) {
      USTax.#instance = new USTax();
    }
    return USTax.#instance;
  }

  getSalesTotal(saleAmount) {
    return saleAmount + (saleAmount * USTax.#TAXRATE);
  }

  getTaxAmount(saleAmount) {
    return saleAmount * USTax.#TAXRATE;
  }

  getTaxRate() {
    return USTax.#TAXRATE;
  }
}

module.exports = USTax;

File: hw/bonilla/U3/HW18Singleton/Singleton/src/main/java/ec/edu/espe/singleton/model/USTax.java
package ec.edu.espe.singleton.model;

/**
 * 
 * @author David Bonilla SoftCrafters ESPE
 */
public class USTax {
    
    private static USTax instance;
    private static final double TAXRATE = 0.15;
    
    private USTax() {}
    
    public static USTax getInstance() {
        if (instance == null) {
            instance = new USTax();
        }
        return instance;
    }
    
    public double getSalesTotal(double saleAmount) {
        return saleAmount + (saleAmount * TAXRATE);
    }
    
    public double getTaxAmount(double saleAmount) {
        return saleAmount * TAXRATE;
    }
    
    public double getTaxRate() {
        return TAXRATE;
    }
}
